src/Compute/Compute.Management.Sdk/README.md

```diff
@@ -0,0 +1,82 @@
+# Overall
+This directory contains the service clients of Az.Compute module.
+
+## Run Generation
+In this directory, run AutoRest:
+```
+autorest.cmd README.md --version=v2
```

Comment:
```suggestion
autorest --reset
autorest --use:@microsoft.azure/autorest.csharp@2.3.90
autorest.cmd README.md --version=v2
```

Reply:
i generated using your autorest commands, and saw the differences. But the SDK package that we are referencing currently and is migrating is using the original way (autorest.cmd README.md --version=v2) 
See here for RollingUpgradePolicy.cs file: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/RollingUpgradePolicy.cs#L166C12-L169C14

Reply:
@vidai-msft

Reply:
@grizzlytheodore Do you mean using above 3 commands to generate the SDK is not correct? This is how we verify the generated files in the CI.

Reply:
the code generated using the 3 commands above is different than the SDK package that is published today.

Using autorest.cmd README.md --version=v2, I get the same code as SDK that is GA. 

@vidai-msft

Reply:
@grizzlytheodore Can you please try with this command autorest --use:@microsoft.azure/autorest.csharp@2.3.90 --version=v2 and see if the generated code is the same as the SDK? In addition, can you please help confirm if the JSON files listed in the README.md are the same as GAed SDK? Thanks!

Reply:
@grizzlytheodore Can you please re-run the `autorest.cmd README.md --version=v2` and then share all the output on the console?

Reply:
The current GA'd SDK uses the same Swagger input files: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/readme.md#tag-package-2023-01-02
and they are generated using: 
`cmd.exe /c autorest.cmd "C:\repositories\swagger\specification\compute\resource-manager\readme.md" --csharp --version=2.0.4421 --reflect-api-versions --csharp-sdks-folder="C:\repositories\sdk\sdk"`

Reply:
@grizzlytheodore Can you share the output from the console when running `autorest.cmd README.md --version=v2`. We want to know what exact version it is using from your side.

Reply:
Here is the output. This produces the code that is a little different from SDK that is GA'd 

`PS C:\Repositories\ps\src\Compute\Compute.Management.Sdk> autorest.cmd README.md --version=v2
AutoRest code generation utility [cli version: 3.6.1; node: v18.13.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: v2.

There is a new version of AutoRest available (3.6.3).
 > You can install the newer version with with npm install -g autorest@latest

   Loading AutoRest core      'C:\Users\thchan\.autorest\@microsoft.azureautorest-core@2.0.4421\nodemodules\@microsoft.azure\autorest-core\dist' (2.0.4421)
   Loading AutoRest extension '@microsoft.azure/autorest.csharp' (~2.3.79->2.3.90)
   Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.55->2.3.55)`

Reply:
@grizzlytheodore Can you share the output that can generate exactly the same code as the GA'd SDK?

Reply:
before i ran,
```
autorest --reset
autorest --use:@microsoft.azure/autorest.csharp@2.3.90
```

`autorest.cmd README.md --version=v2`
would generate the same code as GA'd SDK. 

Now I do: 
```
PS C:\Repositories\ps\src\Compute\Compute.Management.Sdk> autorest --version=2.0.4421
AutoRest code generation utility [cli version: 3.6.1; node: v18.13.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: 2.0.4421.

There is a new version of AutoRest available (3.6.3).
 > You can install the newer version with with npm install -g autorest@latest

   Loading AutoRest core      'C:\Users\thchan\.autorest\@microsoft.azureautorest-core@2.0.4421\nodemodules\@microsoft.azure\autorest-core\dist' (2.0.4421)
   Installing AutoRest extension '@microsoft.azure/autorest.csharp' (~2.3.79)
   Installing AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.55) 
```

Reply:
@grizzlytheodore Per my investigation, the difference is due to the upgrade of autorest.csharp a few months ago. If you compare the code, you would notice that the code generated by 2.3.90 has more if conditions to verify whether the variable is null first because its type is nullable. So we believe that the latest 2.3.101 may have some potential issue. We are now communicating with autorest.csharp team to confirm. Will get back to you.

Reply:
@grizzlytheodore autorest.csharp team has rolled out a new version 2.3.102 to fix the issue. However, Compute SDK seems based on the version 2.3.82 so that it won't get updated even re-generated. As mentioned previously, if you compare the code difference, you will easily find that it makes more sense to verify if a nullable variable is null before checking its real value. So with that, I believe it is safe to generate the code using the updated version. Let me know if you have any concern.

Reply:
makes sense i have no concern

Reply:
@vidai-msft  I've updated the autorest cmd in the README.md to your suggestion and generated following that